libraries/AP_Frsky_Telem/AP_Frsky_SPort_Passthrough.cpp

```diff
@@ -521,6 +522,8 @@ uint32_t AP_Frsky_SPort_Passthrough::calc_ap_status(void)
     ap_status |= (uint8_t)(AP_Notify::flags.failsafe_battery)<<AP_BATT_FS_OFFSET;
     // bad ekf flag
     ap_status |= (uint8_t)(AP_Notify::flags.ekf_bad)<<AP_EKF_FS_OFFSET;
+    //throttle
+    ap_status |= (uint8_t)(gcs().get_hud_throttle())<<AP_THROTTLE_OFFSET;
```

Comment:
This is blindly casting an `int16_t` into a `uint8_t`.  If we can convey a negative value here we should, otherwise it will have to `abs()`

Reply:
@peterbarker  I admit it's a bit rude..and you're right, rover uses negative throttle value even if mavlink VFR_HUD only allows for positive throttle, now what would be the right way, mimic vfr_hud or use an extra bit?
We might also scale 0-100 to 0-64 and use 6+1 bit for sign (allowing for a small error), what do you think?